FFXIVClientStructs/FFXIV/Client/UI/Misc/UIInputData.cs

```diff
@@ -0,0 +1,160 @@
+    [FieldOffset(0xE8)] public float R2;
+    
+    /*
+     * UIFiltered means those are not set if 
+     * - the game window is focused and
+     * - the cursor is hovering any interactable UI elements or windows
+     * 
+     * For mouse buttons, only Left and Right buttons are filtered out, extra buttons are not
+     */
+    [FieldOffset(0x498)] public int UIFilteredCursorXPosition;
+    [FieldOffset(0x49C)] public int UIFilteredCursorYPosition;
+    [FieldOffset(0x4A0)] public int UIFilteredMouseWheel; // -1 for scroll down, 1 for scroll up
+    [FieldOffset(0x4A4)] public MouseButtonFlags UIFilteredMouseButtonHeldFlags;
+    [FieldOffset(0x4A8)] public MouseButtonFlags UIFilteredMouseButtonPressedFlags;
+    [FieldOffset(0x4B0)] public MouseButtonFlags UIFilteredMouseButtonReleasedFlags;
+    [FieldOffset(0x4B4)] public MouseButtonFlags UIFilteredMouseButtonHeldThrottledFlags;
+    [FieldOffset(0x4B8)] public int UIFilteredCursorXDelta; // Delta since last frame
+    [FieldOffset(0x4BC)] public int UIFilteredCursorYDelta; // Delta since last frame
+
+    // Same as 0x4F4 ?
+    // [FieldOffset(0x4C4)] public byte IsGameWindowFocused;
+
+    [FieldOffset(0x4C8)] public int CursorXPosition;
+    [FieldOffset(0x4CC)] public int CursorYPosition;
+    [FieldOffset(0x4D0)] public int MouseWheel; // -1 for scroll down, 1 for scroll up
+    [FieldOffset(0x4D4)] public MouseButtonFlags MouseButtonHeldFlags;
+    [FieldOffset(0x4D8)] public MouseButtonFlags MouseButtonPressedFlags;
+    [FieldOffset(0x4E0)] public MouseButtonFlags MouseButtonReleasedFlags;
+    [FieldOffset(0x4E4)] public MouseButtonFlags MouseButtonHeldThrottledFlags;
+    [FieldOffset(0x4E8)] public int CursorXDelta; // Delta since last frame
+    [FieldOffset(0x4EC)] public int CursorYDelta; // Delta since last frame
+
+    // At least this is what it seems to be
+    [FieldOffset(0x4F4)] public bool IsGameWindowFocused;
+
+    /*
+     * All the following keyboard keys states are not triggered if the chat input is active
+     */
+
+    /*
+     * This one is weird. Seems to fire as long as the last key pressed is still held
+     * Except modifiers (ctrl, shift, alt) where it only fires once
+     * If fires less often than the "Down" flag in the array below but more often than the "Held" one
+     * So Im not sure what to make of this or if this is even useful
+     */
+    //[FieldOffset(0x4F8)] public int IsLastKeyboardKeyDownThrottled;
+
+    [FixedSizeArray<KeyStateFlags>(133)]
+    [FieldOffset(0x51C)] public fixed int KeyState[0x4 * 133];
+    public KeyStateFlags GetKeyState(int key)
+    {
+        int offset;
+        if (key == 144) // VirtualKey.NUMLOCK
+            offset = 0x1E0;
+        else if (key == 145) // VirtualKey.SCROLL
+            offset = 0x1E4;
+        else
+            offset = key - 8;
+
+        if (offset < 0 || offset > 133)
+            throw new ArgumentOutOfRangeException();
+
+        return (KeyStateFlags)KeyState[offset];
```

Comment:
FYI, since you added [FixedSizeArray<KeyStateFlags>] on the field, you could also just do:
return KeyStateSpan[offset];

It's not substantially different here, just want you to be aware :-)